### PR TITLE
added demonstration: RGB to HSV and thresholding

### PR DESCRIPTION
**Pull request by aquatiko**
This demo shows how to convert RGB to HSV and threshold images based on their respective hue, saturation and value.

### CODE REVIEW

File: docs/src/demos/rgb_hsv_thresholding.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+A simple segmentation of the image can then be effectively performed 
+by a mere thresholding of the HSV channels.
+
+
+```julia
+using Images, TestImages, LinearAlgebra, Interact, ImageView
+
+rbg_img = testimage("lighthouse")
+hsv_img = HSV.(rbg_img)
+channels = float(channelview(img))
```
Comment (Evizero) [inline docs/src/demos/rgb_hsv_thresholding.md]: what is `img` ? should this be `hsv_img`?

File: docs/src/demos/rgb_hsv_thresholding.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+hue_img = channels[1,:,:]
+value_img = channels[3,:,:]
+saturation_img = channels[2,:,:]
+@manipulate for hue in 0:255, saturation in 0:255, value in 0:255
+    binary_img = zeros(size(hue_img))
```
Comment (Evizero) [inline docs/src/demos/rgb_hsv_thresholding.md]: can this be preallocated outside of `@manipulate` or does that break the code somehow?

Comment (aquatiko) [reply docs/src/demos/rgb_hsv_thresholding.md]: @Evizero It won't work outside, the `binary_img` will change after each thresholding and the array needs to be reset each time.

Comment (Evizero) [reply docs/src/demos/rgb_hsv_thresholding.md]: right. I was more talking about moving the allocation outside and using `fill!(binary_img, 0.0)` inside instead (which doesn't allocate)

Comment (aquatiko) [reply docs/src/demos/rgb_hsv_thresholding.md]: @Evizero I wasn't quite aware the method. I pushed the suggested changes. 
Just a question out of curiosity,
Does this speeds up the process or is it according to coding standards?

Comment (Evizero) [reply docs/src/demos/rgb_hsv_thresholding.md]: well `zeros()` allocates a new array every time it is called and then fills that array with zeros.
this new approach reuses the same array every time and just resets the values. In pretty much all cases that i encountered this kind of preallocation will be faster

**Comment by Evizero** [comment]
LGTM as far as I can judge

I do think though this needs to be linked to from somewhere. probably https://github.com/JuliaImages/juliaimages.github.io/blob/source/docs/src/demos.md (right @timholy ?)

**Comment by aquatiko** [comment]
@Evizero Yes, I forgot about that. Will add it soon.
I saw on the idea page of **GSoC(Julia)** about adding demos as a project.
I am a bit confused if all the demos needs to be GUI supported or should they just show different things and techniques we can use from JuliaImages?

**Comment by Evizero** [comment]
> I am a bit confused if all the demos needs to be GUI supported or should they just show different things and techniques we can use from JuliaImages?

The later I am pretty sure. Maybe Prof. Holy has something to add to that

**Comment by Evizero** [comment]
can you rebase your commits on the current `source` branch? I made a force push to fix some things

**Comment by aquatiko** [comment]
@Evizero Can this be merged now?

**Comment by Evizero** [comment]
thanks!

**Pull request merged**